src/tests/DataLoopbackBufferTest/DataLoopbackBufferTester.py

```diff
@@ -0,0 +1,198 @@
+                        index, size, hex(element)
+                    )
+                )
+
+                print("dut output:")
+                print(
+                    "state_index:{} \nstate_size:{} \nstate_element:{}".format(
+                        dut_index, dut_size, hex(dut_element)
+                    )
+                )
+                print(
+                    "state_id:{}".format(int(dut.io_outputs_1_payload_state_id.value))
+                )
+                raise TestFailure("test case {} failed".format(cases_count))
+
+        if (dut.io_outputs_0_ready.value & dut.io_outputs_0_valid.value) == True:
+            cases_count, size, index, element = ref_outputs.get()
+            dut_size = int(dut.io_outputs_0_payload_state_size.value)
+            dut_element = int(dut.io_outputs_0_payload_state_element.value)
+            dut_index = int(dut.io_outputs_0_payload_state_index.value)
+            if (dut_size != size) | (dut_element != element) | (dut_index != index):
+                print("cases_num:{}".format(cases_count))
+                print("reference output:".format(cases_count))
+                print(
+                    "state_index:{} \nstate_size:{} \nstate_element:{}".format(
+                        index, size, hex(element)
+                    )
+                )
+
+                print("dut output:".format(cases_count))
+                print(
+                    "state_index:{} \nstate_size:{} \nstate_element:{}".format(
+                        dut_index, dut_size, hex(dut_element)
+                    )
+                )
+                print(
+                    "state_id:{}".format(int(dut.io_outputs_0_payload_state_id.value))
+                )
+                raise TestFailure("test case {} failed".format(cases_count))
+
+        if (cases_count == cases_num) & ref_outputs.empty():
+            raise TestSuccess(" pass {} test cases".format(cases_num))
+
+
+@cocotb.test(timeout_time=200000, timeout_unit="ns")
+async def DataLoopbackBufferTest(dut):
+    await cocotb.start(Clock(dut.clk, 10, "ns").start())
+
+    # set default values to all dut input ports
+    dut.io_input_valid.value = False
+    dut.io_input_payload_round_index.value = 0
+    dut.io_input_payload_state_size.value = 0
+    dut.io_input_payload_state_id.value = 0
+    dut.io_input_payload_state_elements_0.value = 0
+    dut.io_input_payload_state_elements_1.value = 0
+    dut.io_input_payload_state_elements_2.value = 0
+    dut.io_input_payload_state_elements_3.value = 0
+    dut.io_input_payload_state_elements_4.value = 0
+    dut.io_input_payload_state_elements_5.value = 0
+    dut.io_input_payload_state_elements_6.value = 0
+    dut.io_input_payload_state_elements_7.value = 0
+    dut.io_input_payload_state_elements_8.value = 0
+    dut.io_input_payload_state_elements_9.value = 0
+    dut.io_input_payload_state_elements_10.value = 0
+    dut.io_input_payload_state_elements_11.value = 0
```

Comment:
consider better way to assign values to a vector

Reply:
fixed